src/core/thread/mle_router.cpp

```diff
@@ -4278,6 +4292,26 @@ bool MleRouter::HasOneNeighborwithComparableConnectivity(const RouteTlv &aRoute,
     return rval;
 }
 
+bool MleRouter::HasChildren(void)
+{
+    Child *children;
+    uint8_t maxChildCount = 0;
+    bool hasChildren = false;
+
+    children = GetChildren(&maxChildCount);
```

Comment:
nit: you can avoid this call by accessing `mChildren` and `mMaxChildrenAllowed` members directly.